azure-pipelines.infra-core.yaml

```diff
@@ -0,0 +1,61 @@
+resources:
+  repositories:
+    - repository: templates
+      type: github
+      name: hmcts/azure-devops-templates
+      ref: refs/heads/master
+      endpoint: hmcts
+
+trigger:
+  - none
+
+parameters:
+- name: environment
+  type: object
+  default:
+   - Dev
+   - Demo
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+variables:
+- name: tfVersion
+  value: 1.1.0
+- name: tfStateServiceConnection
+  value: DTS-SHAREDSERVICES-PROD-Video Hearings
+- group: external
+
+stages:
+
+################################################
+# Check State. #################################
```

Comment:
do we need a CI stage first?
https://github.com/hmcts/azure-devops-templates/blob/master/templates/tests/terraform/allTests.yaml

Reply:
I left this out for a number of reasons:

1.  terraform validate is run in the Plan stage anyway, so seems pointless having it twice.
2. terraform fmt only formats your code to look nice, which is only going to affect the cloned code on the DevOps agent, it isn't going to impact performance of Terraform, the person who writes the code on their local machine should run that anyway after they have completed work and then push to git.
3.  Why are we validating file size?
4. TF Sec was never working (see below)

I don't think we need another stage, if we do want to get TFSEC, or continue to validate file size, I believe we should add this to the plan stage as extra steps.